core/tee/tee_obj.c

```diff
@@ -53,6 +53,25 @@ TEE_Result tee_obj_get(struct tee_ta_ctx *ctx, uint32_t obj_id,
 	return TEE_ERROR_BAD_PARAMETERS;
 }
 
+TEE_Result tee_obj_get1(struct tee_ta_ctx *ctx, uint32_t obj_id,
+		       struct tee_obj **obj)
+{
+	struct tee_obj *o;
+
+	TAILQ_FOREACH(o, &ctx->objects, link) {
+		if (obj_id == (uint32_t) o) {
+			if (o->info.objectType == TEE_TYPE_CORRUPTED_OBJECT) {
+				tee_obj_close(ctx, o);
```

Comment:
I understand your point. I agree that the "close" could be done from libutee. But I have concerns because:
1/ in the specification I have : "The TEE_TYPE_CORRUPTED_OBJECT is used solely in the deprecated TEE_GetObjectInfo". Since TEE_GetObjectInfo/TEE_GetObjectInfo1 use the same TEE_ObjectInfo, I understand that the objectType value is treated at TEE_GetObjectInfo level only (update TEE_ObjectInfo fields) . What do you think ?
2/ Any function that accesses a persistent object handle MAY return a status of TEE_ERROR_CORRUPT_OBJECT or TEE_ERROR_CORRUPT_OBJECT_2, which indicates that corruption of the object has been detected. Before this status is returned, the Implementation SHALL delete the corrupt object and SHALL close the associated handle; subsequent use of the handle SHALL cause a panic.
From my perpective, if I close the object from libutee (I understand it as a subsequent use of the object handle), it will not panic then. What do you think ?